### ISSUE TITLE
[MVCF/Reloading]Harmony patch for ammo generation isn't working

### ISSUE THREAD

**Issue by BBLKepling** [Vanilla-Expanded/VanillaExpandedFramework#101]
TLDR: Change the Reloading Harmony patch for `PawnInventoryGenerator.GenerateInventoryFor` to target `PawnWeaponGenerator.TryGenerateWeaponFor` instead

My mod GNAT Framework was having the same issue where my Harmony patch for `PawnInventoryGenerator.GenerateInventoryFor` wasn't spawning items on newly spawned pawns. After looking over at what Combat Extended was doing I found that they were patching the same method back in 1.4. CE switched their patch over to `PawnWeaponGenerator.TryGenerateWeaponFor` in 1.5, and after switching the GNAT patch over to the same method I got it spawning properly again. My guess is that weapon generation was removed from the 1st method and moved to it's own method causing everyone's code to try to run on pawns without weapons, and changing the Reloading Harmony patch to the new method should have everything back to working order.

Also thanks goes out to Killathon on Discord for helping me sort this much out so my report is more than just "Shit broke, please fix"
